tests/hypothesis/bin_test.py

```diff
@@ -26,3 +26,18 @@ def test_bin_2d_linspace_bounds(a, b):
     y_edges = sc.linspace('y', x.min(), upper, num=3)
     binned = table.bin(x=x_edges, y=y_edges)
     assert binned.sum().value == 2
+
+
+@settings(max_examples=1000)
+@given(st.integers(min_value=0, max_value=5555), st.integers(min_value=0,
+                                                             max_value=1234),
+       st.integers(min_value=0, max_value=1234))
+def test_automatic_grouping_optimization(nrow, i, j):
+    base = sc.DataArray(sc.ones(dims=['row'], shape=[nrow]))
+    base.coords['label'] = sc.arange('row', 0, nrow, dtype='int64')
+    # Use overlapping or non-overlapping sections to simulate labels with duplicates as
+    # well as gaps
+    table = sc.concat([base[:i], base[j:]], 'row')
+    assert sc.identical(
+        table.group('label').coords['label'],
+        sc.array(dims=['label'], values=np.unique(table.coords['label'].values)))
```

Comment:
`table` has each label either once or twice with a majority of once, right? Also, labels are mostly sorted, i.e. it tests a very particular arrangement of inputs.

What is the reason for not just using a random label?

Reply:
If they are random it is difficult to ensure that we see the edges cases: Missing labels in the first section, not full range covered, only full range covered. Unless you have a concrete suggestion?